File: superHeroInfoAppwithJava/app/src/main/java/com/abdbzkn/superheroinfoappwithjava/SuperheroAdapter.java
package com.abdbzkn.superheroinfoappwithjava;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.abdbzkn.superheroinfoappwithjava.databinding.RecyclerRowBinding;
import java.util.ArrayList;
public class SuperheroAdapter extends RecyclerView.Adapter<SuperheroAdapter.SuperheroHolder> {
    ArrayList<Superhero> superheroArrayList;
    public SuperheroAdapter(ArrayList<Superhero> superheroArrayList) {
        this.superheroArrayList = superheroArrayList;
    }
    public class SuperheroHolder extends RecyclerView.ViewHolder{
        private RecyclerRowBinding binding;
        public SuperheroHolder(RecyclerRowBinding binding){
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @NonNull
    @Override
    public SuperheroHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RecyclerRowBinding recyclerRowBinding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
        return new SuperheroHolder(recyclerRowBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull SuperheroHolder holder, int position) {
        holder.binding.recyclerViewTextView.setText(superheroArrayList.get(position).name);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(holder.itemView.getContext(),DetailsActivity.class);
                intent.putExtra("superhero",superheroArrayList.get(position));
                holder.itemView.getContext().startActivity(intent);
            }
        });
    }
    @Override
    public int getItemCount() {
        return superheroArrayList.size();
    }
}

File: superHeroInfoAppwithKotlin/app/src/main/java/com/abdbzkn/superheroinfoappwithkotlin/SuperheroAdapter.kt
package com.abdbzkn.superheroinfoappwithkotlin

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.abdbzkn.superheroinfoappwithkotlin.databinding.RecyclerRowBinding

class SuperheroAdapter (val superheroList : ArrayList<Superhero>): RecyclerView.Adapter<SuperheroAdapter.SuperheroHolder>() {
    class SuperheroHolder(val binding : RecyclerRowBinding) : RecyclerView.ViewHolder(binding.root){
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SuperheroHolder {
       //layout ile bağlama viewBinding kullanarak
        val binding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.context),parent,false)
        return SuperheroHolder(binding)
    }
    override fun getItemCount(): Int {
       //kaç tane olucak
        return superheroList.size
    }
    override fun onBindViewHolder(holder: SuperheroHolder, position: Int) {
        //baplandıktan sonra ne olucak
        holder.binding.recyclerViewTextView.text = superheroList.get(position).name

        holder.itemView.setOnClickListener{
            val intent = Intent(holder.itemView.context,DetailsActivity::class.java)
            MySingleton.chosenSuperhero = superheroList.get(position)
            //intent.putExtra("superhero", superheroList.get(position)) intent ile göndermiycem
            holder.itemView.context.startActivity(intent)
        }
    }
}